lib/sprockets/commoner/bundle.rb

```diff
@@ -34,19 +44,37 @@ class Bundle < Schmooze::Base
       OUTRO = <<-JS.freeze
 }();
 JS
+      def initialize(root)
+        super(root, 'NODE_PATH' => JS_PACKAGE_PATH)
+      end
+
       def self.instance(env)
         @instance ||= new(env.root)
       end
 
       def self.call(input)
-        instance(input[:environment]).call(input)
+        env = input[:environment]
+        instance(env).call(input)
       end
 
       def call(input)
         return unless input[:metadata][:commoner_enabled]
+        env = input[:environment]
+        assets_in_bundle = Set.new(input[:metadata][:included]
+          .map { |uri| env.load(uri) }
+          .map { |asset| env.resolve(asset.filename, type: input[:content_type], pipeline: :self, compat: false)[0] })
+        assets_missing = Set.new(input[:metadata][:commoner_required]) - assets_in_bundle
+
+        global_identifiers = assets_missing.map do |uri|
+          asset = env.load(uri)
+          filename = asset.filename
+          global = asset.metadata[:commoner_global_identifier]
+          raise InaccessibleStubbedFileError, "#{filename} is stubbed in #{input[:filename]} but doesn't define a global. Add an 'expose' directive." if global.nil?
+          [filename.slice(env.root.size + 1, filename.size), global]
```

Comment:
ah yeah! I guess I could do that, but it would bloat the communication between ruby and JS which I'd like to keep lean, as anything that gets sent between them needs to be JSON encoded and decoded